synapse/lib/chop.py

```diff
@@ -127,3 +129,113 @@ def replaceUnicodeDashes(valu):
     for dash in unicode_dashes:
         valu = valu.replace(dash, '-')
     return valu
+
+def cvss2_normalize(vect):
+    '''
+    Helper function to normalize CVSS2 vectors
+    '''
+    vdict = cvss_validate(vect, s_cvss.cvss2)
+    return cvss_normalize(vdict, s_cvss.cvss2)
+
+def cvss3x_normalize(vect):
+    '''
+    Helper function to normalize CVSS3.X vectors
+    '''
+    vdict = cvss_validate(vect, s_cvss.cvss3_1)
+    return cvss_normalize(vdict, s_cvss.cvss3_1)
+
+def cvss_normalize(vdict, vers):
+    '''
+    Normalize CVSS vectors
+    '''
+    metrics = s_cvss.metrics[vers]
+    undefined = s_cvss.undefined[vers]
+
+    vals = []
+    for key in metrics:
+        if key in vdict and vdict[key] != undefined:
+            vals.append(f'{key}:{vdict[key]}')
```

Comment:
We can rewrite this into a single dictionary lookup on `vdict` 

```
    for key in metrics:
        valu = vdict.get(key, undefined)
        if valu != undefined:
            vals.append(f'{key}:{valu}')
```

Reply:
Great suggestion, thanks!